app/notification.go

```diff
@@ -230,6 +252,10 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 					mac <- model.NewAppError("SendNotifications", "app.channel.autofollow.app_error", nil, err.Error(), http.StatusInternalServerError)
 					return
 				}
+				followersMutex.Lock()
+				followers = append(followers, userID)
+				followersMutex.Unlock()
+
```

Comment:
With this addition a user who has participated in a thread and then unfollowed the thread will get notifications. Is this intended?

Reply:
@ashishbhate this was made to add a newly mentioned user to the followers array.
I think we are exiting early is there was an existing membership and the user is not following
https://github.com/mattermost/mattermost-server/blob/e90d9970466de01d3daed6f6835e2c409c498a3a/app/notification.go#L230-L235
at least that was my impression.